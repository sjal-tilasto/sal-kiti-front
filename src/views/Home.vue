<template>
  <div class="home">
    <b-row>
      <b-col>
        <h1>{{ $t("front.header") }}</h1>
        <p>{{ $t("front.description") }}</p>
      </b-col>
    </b-row>
    <b-row>
      <b-col col md="12" lg="8">
        <h2 class="bg-sal-orange">{{ $tc("event.event", 2) }}</h2>
        <EventList :limit="8" :limited="true" />
      </b-col>
      <b-col>
        <b-row>
          <b-col>
            <h2 class="bg-sal-orange">
              {{ $t("sjal.recurve") }} {{ $t("sjal.ranking") }}
            </h2>
            <StatisticsRanking
              division="recurve"
              :dateStart="dateStart"
              :dateEnd="dateEnd"
              :limit="5"
              :limited="true"
            />
            <b-link
              :to="{
                name: 'statistics-ranking',
                query: { division: 'recurve', year: year }
              }"
              >{{ $t("sjal.full_ranking") }}</b-link
            >
          </b-col>
        </b-row>
        <b-row>
          <b-col>
            <h2 class="bg-sal-orange">
              {{ $t("sjal.compound") }} {{ $t("sjal.ranking") }}
            </h2>
            <StatisticsRanking
              division="compound"
              :dateStart="dateStart"
              :dateEnd="dateEnd"
              :limit="5"
              :limited="true"
            />
            <b-link
              :to="{
                name: 'statistics-ranking',
                query: { division: 'compound', year: year }
              }"
              >{{ $t("sjal.full_ranking") }}</b-link
            >
          </b-col>
        </b-row>
        <b-row>
          <b-col>
            <h2 class="bg-sal-orange">
              {{ $t("sjal.barebow") }} {{ $t("sjal.ranking") }}
            </h2>
            <StatisticsRanking
              division="barebow"
              :dateStart="dateStart"
              :dateEnd="dateEnd"
              :limit="5"
              :limited="true"
            />
            <b-link
              :to="{
                name: 'statistics-ranking',
                query: { division: 'barebow', year: year }
              }"
              >{{ $t("sjal.full_ranking") }}</b-link
            >
          </b-col>
        </b-row>
      </b-col>
    </b-row>
  </div>
</template>

<script>
import EventList from "@/components/EventList.vue";
import StatisticsRanking from "@/components/StatisticsRanking.vue";

/**
 * Front page view
 */
export default {
  name: "HomeView",
  components: {
    EventList,
    StatisticsRanking
  },
  computed: {
    year: function() {
      let date = new Date();
      let year = date.getFullYear();
      if (year <= 2022 && date.getMonth() < 5) {
        return 2020;
      }
      if (date.getMonth() < 10) {
        return date.getFullYear() - 1;
      } else {
        return date.getFullYear();
      }
    },
    dateStart: function() {
      return this.year.toString() + "-10-01";
    },
    dateEnd: function() {
      if (this.year === 2020) {
        return (this.year + 2).toString() + "-04-30";
      }
      return (this.year + 1).toString() + "-09-01";
    }
  },
  created() {
    document.title =
      this.$t("title.front_page") + " | " + this.$t("title.prefix");
  }
};
</script>
