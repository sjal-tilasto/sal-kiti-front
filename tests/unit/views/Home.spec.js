import { shallowMount, createLocalVue } from "@vue/test-utils";
import Home from "@/views/Home.vue";

import BootstrapVue from "bootstrap-vue";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(BootstrapVue);
localVue.use(Vuex);

let state = { user: { is_authenticated: false } };
let store = new Vuex.Store({ state });

describe("Home.vue", () => {
  it("renders content", () => {
    const wrapper = shallowMount(Home, {
      localVue,
      store,
      computed: {
        year() {
          return 2021;
        }
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
