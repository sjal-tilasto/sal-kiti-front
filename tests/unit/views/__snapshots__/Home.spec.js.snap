// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders content 1`] = `
<div
  class="home"
>
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h1>
        front.header
      </h1>
       
      <p>
        front.description
      </p>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      col="true"
      lg="12"
      md="12"
      tag="div"
    >
      <h2
        class="bg-sal-orange"
      >
        event.event
      </h2>
       
      <eventlist-stub
        limit="8"
        limited="true"
      />
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      col="true"
      lg="12"
      md="12"
      tag="div"
    >
      <h2
        class="bg-sal-orange"
      >
        sjal.ranking
      </h2>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <b-row-stub
        tag="div"
      >
        <b-col-stub
          lg="4"
          md="6"
          sm="12"
          tag="div"
        >
          <h3
            class="bg-sal-orange"
          >
            
            sjal.recurve
          
          </h3>
           
          <b-link-stub
            target="_self"
            to="[object Object]"
          >
            sjal.full_ranking Y
          </b-link-stub>
           
          <statisticsranking-stub
            division="y"
            limit="5"
            limited="true"
            rolling="true"
          />
           
          <b-link-stub
            target="_self"
            to="[object Object]"
          >
            sjal.full_ranking N
          </b-link-stub>
           
          <statisticsranking-stub
            division="n"
            limit="5"
            limited="true"
            rolling="true"
          />
        </b-col-stub>
         
        <b-col-stub
          lg="4"
          md="6"
          sm="12"
          tag="div"
        >
          <h3
            class="bg-sal-orange"
          >
            
            sjal.compound
          
          </h3>
           
          <b-link-stub
            target="_self"
            to="[object Object]"
          >
            sjal.full_ranking YT
          </b-link-stub>
           
          <statisticsranking-stub
            division="yt"
            limit="5"
            limited="true"
            rolling="true"
          />
           
          <b-link-stub
            target="_self"
            to="[object Object]"
          >
            sjal.full_ranking NT
          </b-link-stub>
           
          <statisticsranking-stub
            division="nt"
            limit="5"
            limited="true"
            rolling="true"
          />
        </b-col-stub>
         
        <b-col-stub
          lg="4"
          md="6"
          sm="12"
          tag="div"
        >
          <h3
            class="bg-sal-orange"
          >
            
            sjal.barebow
          
          </h3>
           
          <b-link-stub
            target="_self"
            to="[object Object]"
          >
            sjal.full_ranking YV
          </b-link-stub>
           
          <statisticsranking-stub
            division="yv"
            limit="5"
            limited="true"
            rolling="true"
          />
           
          <b-link-stub
            target="_self"
            to="[object Object]"
          >
            sjal.full_ranking NV
          </b-link-stub>
           
          <statisticsranking-stub
            division="nv"
            limit="5"
            limited="true"
            rolling="true"
          />
        </b-col-stub>
         
        <b-col-stub
          lg="4"
          md="6"
          sm="12"
          tag="div"
        >
          <h3
            class="bg-sal-orange"
          >
            
            sjal.longbow
          
          </h3>
           
          <b-link-stub
            target="_self"
            to="[object Object]"
          >
            sjal.full_ranking YLB
          </b-link-stub>
           
          <statisticsranking-stub
            division="ylb"
            limit="5"
            limited="true"
            rolling="true"
          />
           
          <b-link-stub
            target="_self"
            to="[object Object]"
          >
            sjal.full_ranking NLB
          </b-link-stub>
           
          <statisticsranking-stub
            division="nlb"
            limit="5"
            limited="true"
            rolling="true"
          />
        </b-col-stub>
         
        <b-col-stub
          lg="4"
          md="6"
          sm="12"
          tag="div"
        >
          <h3
            class="bg-sal-orange"
          >
            
            sjal.traditional
          
          </h3>
           
          <b-link-stub
            target="_self"
            to="[object Object]"
          >
            sjal.full_ranking YTR
          </b-link-stub>
           
          <statisticsranking-stub
            division="ytr"
            limit="5"
            limited="true"
            rolling="true"
          />
           
          <b-link-stub
            target="_self"
            to="[object Object]"
          >
            sjal.full_ranking NTR
          </b-link-stub>
           
          <statisticsranking-stub
            division="ntr"
            limit="5"
            limited="true"
            rolling="true"
          />
        </b-col-stub>
      </b-row-stub>
    </b-col-stub>
  </b-row-stub>
</div>
`;
